ompi/mpi/java/c/Makefile.am

```diff
@@ -14,7 +16,7 @@
 if OMPI_WANT_JAVA_BINDINGS
 
 # Get the include files that were generated from the .java source files
-AM_CPPFLAGS = -I$(top_builddir)/ompi/mpi/java/java $(OPAL_JDK_CPPFLAGS) -DOPAL_DYN_LIB_SUFFIX=\"$(OPAL_DYN_LIB_SUFFIX)\"
+AM_CPPFLAGS = -I$(top_builddir)/ompi/mpi/java/java $(OPAL_JDK_CPPFLAGS) -DOPAL_DYN_LIB_SUFFIX=\"$(OPAL_DYN_LIB_SUFFIX)\" -DOMPI_LIBDIR=\"$(OMPI_WRAPPER_LIBDIR)\"
```

Comment:
Oh, the `JNI_onLoad()` fires before `MPI_INIT`?  Hmm.  Yes, that does create a quandary.  

Hard-coding the answer at compile time seems like it will someday lead to sadness when someone tries to relocate their Open MPI installation, setenv `OPAL_PREFIX` (or one of the others), and is confused as to why the Java bindings don't work.

I'm not sure what the Right answer is.

BTW, mention @shurickdaryin just so that he is aware of this PR.